src/components/biggive-timeline/biggive-timeline.tsx

```diff
@@ -0,0 +1,170 @@
+import { Component, Prop, Element, h } from '@stencil/core';
+import { brandColour } from '../../globals/brand-colour';
+import { spacingOption } from '../../globals/spacing-option';
+
+@Component({
+  tag: 'biggive-timeline',
+  styleUrl: 'biggive-timeline.scss',
+  shadow: true,
+})
+export class BiggiveTimeline {
+  @Element() host: HTMLBiggiveTimelineElement;
+
+  @Prop() spaceBelow: spacingOption = 0;
+
+  @Prop() textColour: brandColour = 'black';
+
+  @Prop() selectedTextColour: brandColour = 'primary';
+
+  @Prop() navigationHighlightColour: brandColour = 'grey-medium';
+
+  @Prop() selectedNavigationHighlightColour: brandColour = 'primary';
+
+  @Prop() buttonBackgroundColour: brandColour = 'white';
+
+  @Prop() buttonIconColour: brandColour = 'primary';
+
+  @Prop() entryBackgroundColour: brandColour = 'white';
+
+  @Prop() entryHighlightColour: brandColour = 'secondary';
+
+  @Prop() entryDateColour: brandColour = 'black';
+
+  @Prop() entryTitleColour: brandColour = 'primary';
+
+  @Prop() entryTextColour: brandColour = 'black';
+
+  currentTab = 0;
+  tabs: Array<any> = [];
+  children: Array<any> = [];
+
+  componentWillLoad() {
+    this.children = Array.from(this.host.children);
+
+    let tabs = this.tabs;
+
+    this.children.forEach(function (entry) {
+      let year = entry.entryDate.substring(0, 4);
+      entry.entryYear = year;
+      if (!tabs.includes(year)) {
+        tabs.push(year);
+      }
+    });
+
+    this.tabs = tabs;
```

Comment:
This line seems redundant - doesn't it work the same without it? You've got 
` let tabs = this.tabs;` and then `this.tabs = tabs;` with no other assignment to the local `tabs` variable in between - so the local tabs is already equal to the `this.tabs` before line 54. The `push` mutates the content of `tabs`, it doesn't assign a new value to the `tabs` reference.

Reply:
Updated